Source/WebCore/html/HTMLTemplateElement.cpp

```diff
@@ -86,6 +86,27 @@ DocumentFragment& HTMLTemplateElement::content() const
     return *m_content;
 }
 
+const AtomString& HTMLTemplateElement::shadowRootMode() const
+{
+    static MainThreadNeverDestroyed<const AtomString> open("open"_s);
+    static MainThreadNeverDestroyed<const AtomString> closed("closed"_s);
+
+    auto modeString = attributeWithoutSynchronization(HTMLNames::shadowrootmodeAttr);
+    if (equalLettersIgnoringASCIICase(modeString, "closed"_s))
+        return closed;
+    if (equalLettersIgnoringASCIICase(modeString, "open"_s))
+        return open;
+    return nullAtom();
+}
+
+void HTMLTemplateElement::setShadowRootMode(const String& value)
```

Comment:
Looks like this should take an AtomString with the appropriate change in the IDL so that the bindings do not allocates a String unnecessarily.

Reply:
Ah, I totally forgot about that attribute. Fixed.